### DISCUSSION TITLE
Could not find package grumpydicator/firefly-iii ... 5.6.10+

### DISCUSSION THREAD

**Discussion by ddebrecenijr** [firefly-iii/firefly-iii#5674] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I am doing  a fresh install on a hosted Ubuntu 20.04 LEMP stack. Following the documentation, I cannot use the `composer` command to install. Running the command without `--prefer-dist firefly-iii 5.6.10` installs 4.3.6. I see #5357  but I am using PHP 8.0.15 with composer 2.2.5.

### Expected behaviour

Running `composer create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii 5.6.10` installs.

### Debug information

```bash
$ composer create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii 5.6.13
Creating a "grumpydictator/firefly-iii" project at "./firefly-iii"


  [InvalidArgumentException]
  Could not find package grumpydictator/firefly-iii with version 5.6.13 in a version installable using your PHP version, PHP extensions and Composer version.
```

**Comment by JC5** [comment]
What's the output of `php -v`?

**Reply by ddebrecenijr** [reply]
```
$ php -v
PHP 8.0.15 (cli) (built: Jan 29 2022 07:24:52) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.15, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.15, Copyright (c), by Zend Technologies
```

**Reply by JC5** [reply]
Then I don't know the solution. This is usually caused by the terminal version of PHP being different from its Apache counterpart.

You could run the composer command again with `-v` (where it stands for verbal) and it might tell you which dependencies its missing.

**Reply by ddebrecenijr** [reply]
Alright, I'll keep playing around with it. Running with -v doesn't appear to show anything meaningful (?).
```
$ composer create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii 5.6.13 -v
Creating a "grumpydictator/firefly-iii" project at "./firefly-iii"


  [InvalidArgumentException]
  Could not find package grumpydictator/firefly-iii with version 5.6.13 in a version installable using
  your PHP version, PHP extensions and Composer version.


Exception trace:
 () at phar:///usr/local/bin/composer/src/Composer/Command/CreateProjectCommand.php:436
 Composer\Command\CreateProjectCommand->installRootPackage() at phar:///usr/local/bin/composer/src/Composer/Command/CreateProjectCommand.php:207
 Composer\Command\CreateProjectCommand->installProject() at phar:///usr/local/bin/composer/src/Composer/Command/CreateProjectCommand.php:168
 Composer\Command\CreateProjectCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:245
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:835
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:185
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:336
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:131
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:83
 require() at /usr/local/bin/composer:29

create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [<package>] [<directory>] [<version>]
```

**Reply by JC5** [reply]
Try this:

```
composer create-project -vvv grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii 5.6.13
```

**Reply by ddebrecenijr** [reply]
Alright, I was able to figure it out. 

Running `apt list --installed | grep php` showed that I had some 7.4.3 packages installed. I removed all of them, then tried again, but didn't have any luck. Running that same command showed that a lot of my 8.0 packages also got uninstalled, which seems like an odd behavior to me, but re-installing everything with `apt instal php8.0-{bcmath,bz2,intl,gd,mbstring,mysql,zip,fpm.ldap,xml,curl} -y` reinstalled everything and it worked fine.

**Reply by ddebrecenijr** [reply]
So not entirely sure what the problem was, I suppose maybe conflicting php versions or maybe I was missing a package initially in my original install. I didn't see your comment about adding the `-vvv` so was unable to test that out. Thanks for all your help!

**Comment by ddebrecenijr** [comment, marked as answer]
Adding here, as this was my answer (?) to close: https://github.com/firefly-iii/firefly-iii/discussions/5674#discussioncomment-2083029

**Comment by krin-san** [comment]
This issue happened to me as well because some of the required PHP extensions were missing.

Full list of required package extensions can be found here: https://packagist.org/packages/grumpydictator/firefly-iii

FreeBSD packages I have installed:
```bash
# Requirements from https://docs.firefly-iii.org/firefly-iii/installation/self_hosted/
php80 php80-bcmath php80-intl php80-curl php80-zip php80-gd php80-xml php80-mbstring php80-ldap
# MariaDB server + PHP drivers
mariadb103-server php80-pear-MDB2 php80-pear-MDB2_Driver_mysqli
# Modules required to install Compose
php80-phar php80-filter php80-openssl
# Package extensions – caused "Could not find package" error
php80-fileinfo php80-iconv php80-pdo_mysql php80-session php80-simplexml php80-sodium php80-tokenizer php80-xmlwriter
# Dependencies of package dependencies – caused "Your lock file does not contain a compatible set of packages" error
php80-dom # tijsverkoyen/css-to-inline-styles
```
